src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.Win32Unix.cs

```diff
@@ -18,21 +18,21 @@ internal static string GetDataDirectory(IsolatedStorageScope scope)
             // (note that Silverlight used "CoreIsolatedStorage" for a directory name and did not support machine scope)
 
             string dataDirectory = null;
-
+            var option = Environment.SpecialFolderOption.Create;
             if (IsMachine(scope))
             {
                 // SpecialFolder.CommonApplicationData -> C:\ProgramData
-                dataDirectory = Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData);
+                dataDirectory = Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData, option);
             }
             else if (IsRoaming(scope))
             {
                 // SpecialFolder.ApplicationData -> C:\Users\Joe\AppData\Roaming
-                dataDirectory = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
+                dataDirectory = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData, option);
             }
             else
             {
                 // SpecialFolder.LocalApplicationData -> C:\Users\Joe\AppData\Local
-                dataDirectory = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+                dataDirectory = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData, option);
             }
```

Comment:
FWIW, I'd find this whole cascading if easier to understand if it were instead:
```C#
Environment.SpecialFolder specialFolder =
    IsMachine(scope) ? Environment.SpecialFolder.CommonApplicationData : // e.g. C:\ProgramData
    IsRoaming(scope) ? Environment.SpecialFolder.ApplicationData : // e.g. C:\Users\Joe\AppData\Roaming
    Environment.SpecialFolder.LocalApplicationData; // e.g. C:\Users\Joe\AppData\Local
dataDirectory = Environment.GetFolderPath(specialFolder, Environment.SpecialFolderOption.Create);
```

Reply:
perhaps I can also change `Environment.SpecialFolder specialFolder` to `var specialFolder` since we can see from code what the type of specialFolder will be.